docs/pkg/client/events.md

```diff
@@ -0,0 +1,376 @@
+
+
+  q1_conn -.-> q1_eb
+  q2_conn -.-> q2_eb
+
+  sub -.-> q1_conn
+  sub -.-> q2_conn
+```
+
+## Installation
+
+```bash
+go get github.com/pokt-network/poktroll/pkg/client/events
+```
+
+## Features
+
+- **Websocket Connection**: Uses the
+  [Gorilla WebSockets](https://github.com/gorilla/websocket) for implementing the
+  connection interface.
+- **Events Subscription**: Subscribe to chain event messages using a simple query
+  mechanism.
+- **Dialer Interface**: Offers a `Dialer` interface for constructing connections,
+  which can be easily mocked for tests.
+- **Observable Pattern**: Integrates the observable pattern, making it easier to
+  react to chain events.
+- **Generic Replay Client**: Offers a generic typed replay client to listen for
+  specifc events on chain, and handles reconnection and subscription on error,
+  if the `EventsQueryClient` returns an error/is closed.
+
+## Usage (`EventsQueryClient`)
+
+### Basic Example
+
+```go
+ctx := context.Background()
+
+// Creating a new EventsQueryClient with the default, websocket dialer:
+cometWebsocketURL := "ws://example.com"
+evtClient := events.NewEventsQueryClient(cometWebsocketURL)
+
+// Subscribing to a specific event, e.g. newly committed blocks:
+// (see: https://docs.cosmos.network/v0.47/core/events#subscribing-to-events)
+observable := evtClient.EventsBytes(ctx, "tm.event='NewBlock'")
+
+// Subscribe and receive from the observer channel, typically in some other scope.
+observer := observable.Subscribe(ctx)
+
+// Observer channel closes when the context is canceled, observer is
+// unsubscribed, or after the subscription returns an error.
+for eitherEvent := range observer.Ch() {
+    // (see either.Either: https://github.com/pokt-network/poktroll/blob/main/pkg/either/either.go#L3)
+    eventBz, err := eitherEvent.ValueOrError()
+
+    // ...
+}
+```
+
+### Advanced Usage
+
+```go
+// Given some custom dialer & connection implementation, e.g.:
+var (
+    tcpDialer events.Dialer = exampletcp.NewTcpDialerImpl()
+    grcpDialer events.Dialer = examplegrpc.NewGrpcDialerImpl()
+)
+
+// Both TCP and gRPC use the TCP scheme as gRPC uses TCP for its transport layer.
+cometUrl = "tcp://example.com"
+
+// Creating new EventsQueryClients with a custom tcpDialer:
+tcpDialerOpt := events.WithDialer(tcpDialer)
+tcpEvtClient := events.NewEventsQueryClient(cometUrl, tcpDialerOpt)
+
+// Alternatively, with a custom gRPC dialer:
+gcpDialerOpt := events.WithDialer(grcpDialer)
+grpcEvtClient := events.NewEventsQueryClient(cometUrl, grpcDialerOpt)
+
+// ... rest follows the same as the basic example.
+```
+
+### Configuration
+
+- **WithDialer**: Configure the client to use a custom dialer for connections.
+
+## Usage (`EventsReplayClient`)
+
+### Basic Usage
+
+```go
+// Define a query string to provide to the EventsQueryClient
+// See: https://docs.cosmos.network/main/learn/advanced/events#subscribing-to-events
+// And: https://docs.cosmos.network/main/learn/advanced/events#default-events
```

Comment:
I had linked to the v0.47 page specifically as some things around events have changed in v0.50 IIRC. I find myself wanting to also suggest adding a TODO to update this (and the analogous comment in `EventsQueryClient`) but I don't think there's a good place for that in the docs. I've started #240 as a way to capture things like this (and I've already added this one).

```suggestion
// See: https://docs.cosmos.network/v0.47/learn/advanced/events#subscribing-to-events
// And: https://docs.cosmos.network/v0.47/learn/advanced/events#default-events
```